pxr/imaging/hio/fieldTextureData.h

```diff
@@ -103,6 +103,8 @@ class HioFieldTextureData
 ///
 /// A base class to make HioFieldTextureData objects, implemented by plugins.
 ///
+ARCH_PRAGMA_PUSH
+ARCH_PRAGMA_NON_DLL_INTERFACE_CLASS_USED_AS_BASE_FOR_DLL_INTERFACE_CLASS
```

Comment:
FactoryBase is properly adorned. Is the real issue that HioFieldTextureDataFactoryBase doesn't export any symbols of its own? Would removing HIO_API fix it?

Reply:
If the structure is not meant to be DLL Exported, then removing the API decoration would also fix it. The core issue is that you have a DLL data structure inheriting from a non-DLL data structure.

Reply:
Original warning:
/pxr/imaging/hio/image.h(202,52): warning C4275: non dll-interface class 'pxrInternal_v0_23__pxrReserved__::TfType::FactoryBase' used as base for dll-interface class 'pxrInternal_v0_23__
pxrReserved__::HioImageFactoryBase'

Removal of the HIO_API does fix it and is the preferred approach?